### DISCUSSION TITLE
程序入口设置问题+给linker传参问题

### DISCUSSION THREAD

**Discussion by ruiyuanlu** [xmake-io/xmake#3761] [Q&A]
求助：

1. 项目中有多个子项目，每个子项目都有自己的 main 函数和对应的 main.cpp，请问如何调整程序入口点？是在不同的 target(xxx) 中 add_files(subproject_main.cpp) 这种嘛？

2. 如何给 linker 传参，在 VS 中可以调整 linker 的属性，lua 脚本中如何传参？目前的方法是在代码文件中使用 #pragma comment(linker, "xxx" ) 传参，想请问一下有没有不需要这么做的方法，在lua中可以传参吗？

**Comment by waruqi** [comment]
1. 一个 target 对应 main.cpp 不需要特殊处理。
2. add_links

建议先自己看下文档

**Reply by ruiyuanlu** [reply]
我试了下，add_links 好像没反应，add_ldflags 好像可以，但是文档里我没找到 add_ldflags 的用法说明，想请问下这两个有什么区别？

**Comment by waruqi** [comment]
https://xmake.io/#/zh-cn/manual/project_target?id=targetadd_ldflags
https://xmake.io/#/zh-cn/manual/project_target?id=targetadd_links

刚说了，自己看文档，都有。。

**Reply by ruiyuanlu** [reply]
看漏了不好意思，打扰了
